pandas/core/indexes/base.py

```diff
@@ -907,16 +907,26 @@ def astype(self, dtype, copy=True):
         """
 
     @Appender(_index_shared_docs["take"] % _index_doc_kwargs)
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
+    def take(
+        self,
+        indices: Union[AnyArrayLike, Sequence[int]],
```

Comment:
isnt Sequence[int] going to cover the AnyArrayLike case?

Reply:
I think even with `AnyArrayLike` in place of `ArrayLike` we run into the issue mentioned here https://github.com/pandas-dev/pandas/pull/40521#discussion_r598605341

Reply:
this is too complicated and specific an annotation to put *here*, leave this untyped for now, or put it in a typing alias, but then we should enforce it, alt we shouldn't allow non-array-likes here

Reply:
Left untyped for now, now mostly just a doc fix